src/modules/assessments/components/household/HouseholdAssessmentTabPanel.tsx

```diff
@@ -70,23 +77,14 @@ const HouseholdAssessmentTabPanel = memo(
       if (!navigatingAway) return;
       // skip save if it's already happening
       if (formState.saving) return;
+      // skip save if there are errors. user must do an explicit save to resolve error before proceeding
+      if (formState.errors) return;
+
+      // perform background save or submit
       if (assessmentSubmitted) {
-        formRef.current.SubmitIfDirty(true, () => {
-          // TODO: Update tab status to 'error' if error?
-          // console.debug(`Submitted ${clientName}!`);
-          onFormStateChange(enrollmentId, 'saveCompleted');
-        });
+        formRef.current.SubmitIfDirty(false);
       } else {
-        formRef.current.SaveIfDirty(() => {
-          onFormStateChange(enrollmentId, 'saveCompleted');
-          // TODO: Update tab status to 'error' if error?
-          // console.debug(`Saved ${clientName}!`);
-          if (!assessmentId) {
-            // This was a NEW assessment; we need to re-fetch to get it
-            updateTabStatus(AssessmentStatus.Started, id);
-            refetch();
-          }
-        });
+        formRef.current.SaveIfDirty();
```

Comment:
We don't need these weird callbacks anymore, because we specify them in the `onCompletedMutation` defined below. We have a standard callback for form submission, so it will happen the same way whether the form is submitted from an explicit "save" or whether it's from a `SaveIfDirty` background save. This still needs improvement, but hopefully this is moving in the right direction.